src/plugins/intel_cpu/src/nodes/search_sorted.cpp

```diff
@@ -0,0 +1,114 @@
+    right_mode = ss_op->get_right_mode();
+}
+
+bool SearchSorted::isSupportedOperation(const std::shared_ptr<const ov::Node>& op, std::string& errorMessage) noexcept {
+    try {
+        if (!ov::is_type<ov::op::v15::SearchSorted>(op)) {
+            errorMessage = "Only opset15 SearchSorted operation is supported";
+            return false;
+        }
+    } catch (...) {
+        return false;
+    }
+    return true;
+}
+
+void SearchSorted::getSupportedDescriptors() {
+    // Validation is already done in the ov::opset15::SearchSorted.
+}
+
+void SearchSorted::initSupportedPrimitiveDescriptors() {
+    if (!supportedPrimitiveDescriptors.empty())
+        return;
+
+    ov::element::Type inputPrec = getOriginalInputPrecisionAtPort(0);
+    ov::element::Type outputPrec = getOriginalOutputPrecisionAtPort(0);
+
+    addSupportedPrimDesc({{LayoutType::ncsp, inputPrec}, {LayoutType::ncsp, inputPrec}},
+                         {{LayoutType::ncsp, outputPrec}},
+                         impl_desc_type::ref);
+}
+
+bool SearchSorted::created() const {
+    return getType() == Type::SearchSorted;
+}
+
+bool SearchSorted::needPrepareParams() const {
+    return false;
+}
+
+void SearchSorted::executeDynamicImpl(dnnl::stream strm) {
+    execute(strm);
+}
+
+template <typename INPUT_TYPE, typename OUTPUT_TYPE>
+void SearchSorted::executeImpl() {
+    ov::reference::search_sorted<INPUT_TYPE, OUTPUT_TYPE>(getSrcDataAtPortAs<const INPUT_TYPE>(0),
+                                                          getSrcDataAtPortAs<const INPUT_TYPE>(1),
+                                                          getDstDataAtPortAs<OUTPUT_TYPE>(0),
+                                                          ov::Shape{getSrcMemoryAtPort(0)->getStaticDims()},
+                                                          ov::Shape{getSrcMemoryAtPort(1)->getStaticDims()},
+                                                          right_mode);
+}
+
+namespace {
+struct SearchSortedContext {
+    SearchSorted& node;
+};
+}  // namespace
+
+template <typename T>
+struct SearchSorted::SearchSortedExecute {
+    using TInputType = typename std::tuple_element<0, T>::type;
+    using TOutputType = typename std::tuple_element<1, T>::type;
+
+    void operator()(SearchSortedContext& ctx) {
+        ctx.node.executeImpl<TInputType, TOutputType>();
+    }
+};
+void SearchSorted::execute(dnnl::stream strm) {
+    auto inputPrecision = getParentEdgeAt(0)->getMemory().getDesc().getPrecision();
+    auto outputPrecision = getChildEdgeAt(0)->getMemory().getDesc().getPrecision();
+
+    SearchSortedContext ctx = {*this};
+
+#define CASE(OV_TYPE)                                                                           \
+    OV_CASE2(OV_TYPE, ov::element::i64, ov::element_type_traits<OV_TYPE>::value_type, int64_t), \
+        OV_CASE2(OV_TYPE, ov::element::i32, ov::element_type_traits<OV_TYPE>::value_type, int32_t)
+
+    OV_SWITCH(intel_cpu,
```

Comment:
Since OV_SWITCH doesn't check if any condition is satisfied, we need to guarantee inputPrecision and outputPrecision are in supported range. In order to do that lets adjust precision which we use in initSupportedPrimitiveDescriptors. See as as example https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/nodes/scatter_update.cpp#L273-L283.

Reply:
Fixed - as discussed in separate channel.